app/services/SignatureService.js

```diff
@@ -0,0 +1,104 @@
+const request = require("request");
+const tmp = require("tmp");
+const fs = require("fs");
+const slug = require("slug");
+const {
+  BACKGROUND_IMAGE_EMPTY_PATH,
+  BACKGROUND_IMAGE_ALLIANCE_PATH,
+  BACKGROUND_IMAGE_HORDE_PATH,
+  BACKGROUND_IMAGE_NEUTRAL_PATH,
+  FONT_MERRIWEATHER_BOLD_PATH,
+  FONT_MERRIWEATHER_REGULAR_PATH
+} = require("../constants");
+
+/**
+ * imageMagick made available by running application in Docker
+ */
+const gm = require("gm").subClass({ imageMagick: true });
+
+class SignatureService {
+
+  async downloadCharacterMediaAsset(mediaUrl) {
+    const tmpName = `${tmp.tmpNameSync()}.png`;
+    return new Promise((resolve, reject) => {
+      request(mediaUrl)
+        .pipe(fs.createWriteStream(tmpName))
+        .on("finish", () => {
+          resolve(tmpName);
+        })
+        .on("error", (err) => {
+          reject(err);
+        });
+    });
+  }
+
+  getBackgroundImagePath(factionEnum) {
+    return factionEnum === "ALLIANCE"
+      ? BACKGROUND_IMAGE_ALLIANCE_PATH
+      : factionEnum === "HORDE"
+        ? BACKGROUND_IMAGE_HORDE_PATH
+        : BACKGROUND_IMAGE_NEUTRAL_PATH;
+  }
+
+  buildInfoString(character) {
+    return `Level ${character.level} ${character.character_class.name} ${
+      character.guild ? `of <${character.guild.name}> ` : ""
+      }on ${character.realm.name}`;
+  }
+
+  async generateImage(character, characterMedia) {
+    const { faction } = character;
+    const { inset: bustUrl } = characterMedia;
+    const backgroundImage = this.getBackgroundImagePath(faction.type);
+    const tmpBustPath = await this.downloadCharacterMediaAsset(bustUrl);
+    const identityString = this.buildInfoString(character);
+    const itemLevelString = `Item Level: ${character.equipped_item_level} (${character.average_item_level})`;
+    const achievementPointsString = `Achievement Points: ${character.achievement_points}`;
+
+    console.log(BACKGROUND_IMAGE_EMPTY_PATH)
```

Comment:
Either remove presumably-temporary logging, or add context if this is intended as a debug statement.